### PR TITLE
`tk-search` - Added unit tests

### PR DESCRIPTION
**Pull request by vnazarchukGD**

### CODE REVIEW

File: src/app/components/search/search.component.html [hunk 1]
```diff
@@ -1,38 +1,42 @@
-<div class='search'>
-  <mat-form-field appearance='fill' class='search__input'>
-    <mat-label i18n='Product search label@@product-search.label'>
+<div class="search">
```
Comment (ooow) [inline src/app/components/search/search.component.html:1]: Please revert this file, single quotes are preferable.

File: src/app/components/search/search.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  template:
+    '<tk-search [options]="options" (valueChange)="valueChange($event)"></tk-search>'
+})
+class TestComponent {
+  options: string[];
+
+  valueChange(event): void {}
+}
+
+describe('AppComponent => SearchComponent', () => {
```
Comment (ooow) [inline src/app/components/search/search.component.spec.ts:36]: Define only the component which is being tested. In this case, is **'SearchComponent'**

File: src/app/components/search/search.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+@Component({
+  selector: 'tk-test',
+  template:
+    '<tk-search [options]="options" (valueChange)="valueChange($event)"></tk-search>'
+})
+class TestComponent {
+  options: string[];
+
+  valueChange(event): void {}
```
Comment (ooow) [inline src/app/components/search/search.component.spec.ts:33]: And again, don't create real methods when you are using _**spyOn**_ instead of them.

**Review by ooow — APPROVED** [review]
LGTM

File: src/app/components/search/search.component.html (hunk 1 above)
Comment (vnazarchukGD) [reply src/app/components/search/search.component.html:1]: are you sure? 
Never heard that single quotes preferable in *.html files, and all Google examples that involve *.html files use double quotes in them.

Comment (ooow) [reply src/app/components/search/search.component.html:1]: Yes, my bad, misplaced the comment.

**Pull request merged**